util/util.cpp

```diff
@@ -66,10 +70,67 @@ double getSensorDbusTemp(std::string sensorDbusPath, bool unitMilli)
         return value;
     }
 
+    if (spofEnabled)
+    {
+        // If sensor functional property is false, set NaN to dbus and return NaN.
+        if (!dbus::SDBusPlus::checkFunctionalProperty(bus,
+                                                      service,
+                                                      sensorDbusPath))
+        {
+            dbus::SDBusPlus::setValueProperty(bus,
```

Comment:
Interesting, does this overwrite the source sensor with NaN if functional is false? I thought read-margin-temp only wrote the values of target sensors it is configured to use as output (the computed worst-margin virtual sensors). Overwriting the source sensors as well would be a change in behavior. I don't use functional, but if you feel this is OK, then it's OK, but I just wanted to point this out.

Reply:
Because when functional is false, the value property will not update and keep the value in last update.
The dbus-sensor will set [NaN](https://github.com/openbmc/dbus-sensors/blob/master/include/sensor.hpp#L344) to value property. But in phosphor-hwmon, it just use [continue](https://github.com/openbmc/phosphor-hwmon/blob/master/mainloop.cpp#L440) to skip the value update. That's why I want to set NaN to sensor value property in this place.
How do you think? Do I need to add this function when funtional is false?

Reply:
That is interesting. Our application does not use phosphor-hwmon, it uses dbus-sensors exclusively. It almost seems like a bug in phosphor-hwmon, that it does not clear the old stale sensor value to NaN when there is a sensor error. The behavior of phosphor-hwmon and dbus-sensors is different here, which seems like a bug to me.

A better approach might be to fix this bug in phosphor-hwmon, instead of trying to do it here in read-margin-temp. Or, make this a configurable option, so that users can disable the NaN behavior if their application can not handle NaN and they would prefer to have stale readings survive, instead of clearing them to NaN here in read-margin-temp.

Reply:
OK, I see. I will remove this function in this place.

Reply:
Done.